DSharpPlus.UnifiedCommands/Internals/CommandModuleRegister.cs

```diff
@@ -0,0 +1,344 @@
+                            if (parameter.ParameterType != typeof(string))
+                            {
+                                throw new InvalidMessageModuleStructure(
+                                    "You need to use string if you are using RemainingTextAttribute.");
+                            }
+
+                            parameterData.WillConsumeRestOfArguments = true;
+                            cantConsumeMoreArguments = true;
+                        }
+                    }
+                    else
+                    {
+                        parameterData.IsPositionalArgument = false;
+                        parameterData.Name = paramAttribute.Option;
+                        parameterData.ShorthandOptionName = paramAttribute.ShorthandOption;
+                    }
+
+
+                    parameterData.CanBeNull =
+                        nullabilityContext.Create(parameter).WriteState is NullabilityState.Nullable;
+                    Type parameterType =
+                        Nullable.GetUnderlyingType(parameter.ParameterType) ?? parameter.ParameterType;
+                    parameterData.HasDefaultValue = parameter.HasDefaultValue;
+
+                    // I don't know how I would make this into a switch case. 
+                    if (parameterType == typeof(string))
+                    {
+                        parameterData.Type = MessageParameterDataType.String;
+                    }
+                    else if (parameterType == typeof(int))
+                    {
+                        parameterData.Type = MessageParameterDataType.Int;
+                    }
+                    else if (parameterType == typeof(DiscordUser))
+                    {
+                        parameterData.Type = MessageParameterDataType.User;
+                    }
+                    else if (parameterType == typeof(DiscordRole))
+                    {
+                        parameterData.Type = MessageParameterDataType.Role;
+                    }
+                    else if (parameterType == typeof(bool))
+                    {
+                        parameterData.Type = MessageParameterDataType.Bool;
+                    }
+                    else if (parameterType == typeof(DiscordMember))
+                    {
+                        parameterData.Type = MessageParameterDataType.Member;
+                    }
+                    else if (parameterType == typeof(DiscordChannel))
+                    {
+                        parameterData.Type = MessageParameterDataType.Channel;
+                    }
+                    else
+                    {
+                        parameterData.Type = parameterType == typeof(double)
+                            ? MessageParameterDataType.Double
+                            : throw new InvalidMessageModuleStructure(
+                                                    $"You cannot use type {parameterType.Name} as a parameter.");
```

Comment:
Would be nice if the error message contains the methodname, so the user can see easily where they have to change something